src/ol/TileQueue.js

```diff
@@ -112,12 +112,10 @@ class TileQueue extends PriorityQueue {
       if (state === TileState.ABORT) {
         abortedTiles = true;
       } else if (state === TileState.IDLE && !(tileKey in this.tilesLoadingKeys_)) {
+        this.tilesLoadingKeys_[tileKey] = true;
+        ++this.tilesLoading_;
+        ++newLoads;
         tile.load();
-        if (tile.getState() === TileState.LOADING) {
-          this.tilesLoadingKeys_[tileKey] = true;
-          ++this.tilesLoading_;
-          ++newLoads;
-        }
```

Comment:
Reviewing this last time, I liked that we would never count a tile as loading if the state wasn’t loading.  I trust that this change improves things, but I admit I can’t explain why we would increment the count if the state is not loading.  

Maybe the new logic updates the state after incrementing the count.